utils/build/docker/nodejs/nextjs/src/app/tag_value/[tag]/[status]/route.js

```diff
@@ -8,6 +8,38 @@ export async function GET (request, { params }) {
 
   rootSpan?.setTag('appsec.events.system_tests_appsec_event.value', params.tag)
 
-  return NextResponse.json('Value tagged')
+  // read querystring
+  const query = Object.fromEntries(new URL(request.url).searchParams)
+
+  // always read body for API sec
+  const body = await getBody(request)
+
+  let response
+
+  if (params?.tag?.startsWith?.('payload_in_response_body') && request.method === 'POST') {
+    response = { payload: body }
+  } else {
+    response = 'Value tagged'
+  }
+
+  return NextResponse.json(response, {
+    status: params.status || 200,
+    headers: new Headers(query)
+  })
 }
 export const OPTIONS = GET
+export const POST = GET
+
+// this function works but only when formData() is called before json()
+// because formData doesn't try to read the body if it's not has form data headers
```

Comment:
```suggestion
// this function only works when formData() is called before json()
// because formData doesn't consume the body if it doesn't have form data headers
```

Reply:
`it's not has` I don't know wtf I was thinking when I wrote that🤔